### PR TITLE
Realigning Build Information resources

### PR DESCRIPTION
**Pull request by slewis74**
The client resources for Deployment Changes haven't stayed in alignment with those that Octopus server is actually using, this brings them back into alignment.

Relates to https://github.com/OctopusDeploy/Issues/issues/7425

### CODE REVIEW

File: source/Octopus.Server.Client/Model/BuildInformation/OctopusPackageVersionBuildInformationMappedResource.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ public class OctopusPackageVersionBuildInformationMappedResource : Resource
 
         public string IssueTrackerName { get; set; }
         public WorkItemLink[] WorkItems { get; set; }
-        public CommitDetail[] Commits { get; set; }
+        public CommitDetails[] Commits { get; set; }
```
Comment (johnsimons) [inline source/Octopus.Server.Client/Model/BuildInformation/OctopusPackageVersionBuildInformationMappedResource.cs:22]: @slewis74 Do these require a major bump?

File: source/Octopus.Server.Client/Model/IssueTrackers/CommitDetails.cs [hunk 2]
```diff
@@ -1,11 +1,9 @@
-namespace Octopus.Client.Model
+namespace Octopus.Client.Model.IssueTrackers
```
Comment (johnsimons) [inline source/Octopus.Server.Client/Model/IssueTrackers/CommitDetails.cs:1]: Breaking change

**Review by johnsimons — APPROVED** [review]
@slewis74 these seem to be non-backwards changes, so are we planning to bump major?
Regardless, changes look fine to me 👍

File: source/Octopus.Server.Client/Model/BuildInformation/OctopusPackageVersionBuildInformationMappedResource.cs (hunk 1 above)
Comment (slewis74) [reply source/Octopus.Server.Client/Model/BuildInformation/OctopusPackageVersionBuildInformationMappedResource.cs:22]: yeah, will be a major bump

**Pull request merged**